src/Tables.jl

```diff
@@ -587,11 +587,37 @@ or an independent copy:
 - If `view=true` then a view is returned and if `view=false` a copy is returned.
   This applies both to returning a row or a table.
  
-Any specialized implementation of `getrows` must support the `view=nothing` argument. 
+Any specialized implementation of `subset` must support the `view=nothing` argument.
 Support for `view=true` or `view=false` is optional
 (i.e. implementations might error on them if they are not supported).
 """
-function getrows end
+function subset(x::T, inds; view::Union{Bool, Nothing}=nothing) where {T}
+    # because this method is being called, we know `x` didn't define it's own Tables.subset
+    # first check if it supports column access, and if so, apply inds and wrap columns in a DictColumnTable
+    if columnaccess(x)
+        cols = columns(x)
+        if inds isa Integer
+            return ColumnsRow(cols, inds)
+        else
+            ret = view === true ? _map(c -> Base.view(c, inds), cols) : _map(c -> c[inds], cols)
+            return DictColumnTable(schema(cols), ret)
```

Comment:
does `DictColumnTable` check that it gets vectors as columns?

Reply:
As a special case someone can pass `CartesianIndex` as `inds` - do we want to handle it in the `ColumnsRow` case or error on it?